src/QsCompiler/CompilerOptimizations/ConstantPropagation.fs

```diff
@@ -27,101 +29,129 @@ type ConstantPropagator(compiledCallables: ImmutableDictionary<QsQualifiedName,
-                let iterVals = statementKind.ExpressionTransformation stm.IterationValues
-                let iterValsAsSeq =
-                    match iterVals.Expression with
-                    | RangeLiteral _ -> 
-                        rangeLiteralToSeq iterVals.Expression |> Seq.map (fun x -> wrapExpr (IntLiteral x) Int) |> Some
-                    | ValueArray va -> va :> seq<_> |> Some
-                    | _ -> None
-
-                match iterValsAsSeq with
-                | Some s ->
-                    let iterRange = 
-                        s |> Seq.map (fun x ->
-                            let variableDecl = QsBinding.New ImmutableBinding (loopVar, x) |> QsVariableDeclaration
-                            let variableDeclStatement = {
-                                Statement = variableDecl
-                                SymbolDeclarations = stm.Body.KnownSymbols
-                                Location = Null
-                                Comments = QsComments.New ([], []) }
-                            let innerScope =
-                                { stm.Body with 
-                                    Statements = stm.Body.Statements.Insert(0, variableDeclStatement) }
-                            {   Statement = innerScope |> QsScopeStatement.New |> QsScopeStatement
-                                SymbolDeclarations = stm.Body.KnownSymbols
-                                Location = Null
-                                Comments = QsComments.New ([], []) })
-                    let outerScope = QsScope.New (iterRange, stm.Body.KnownSymbols)
-                    let result = outerScope |> QsScopeStatement.New |> QsScopeStatement |> statementKind.Transform
-                    // printfn "Unrolled for loop!\nOld statement was:\n%O\nNew statement is:\n%O" (printStm 0 (QsForStatement stm)) (printStm 0 result)
-                    result
-                | None ->
-                    let loopVarType = statementKind.TypeTransformation (snd stm.LoopItem)
-                    let body = statementKind.ScopeTransformation stm.Body
-                    QsForStatement.New ((loopVar, loopVarType), iterVals, body) |> QsForStatement
         }
     }
+
+
+/// The StatementKindTransformation used to simplify statements
+and [<AbstractClass>] private StatementOptimizer(stateRef: TransformationState ref) =
+    inherit StatementKindTransformation()
+
+
+    override this.onExpressionStatement ex =
+        let ex = this.ExpressionTransformation ex
+        maybe {
+            let! qualName, newScope = tryInline !stateRef ex
+            let mySet = HashSet()
+            findAllCalls !stateRef newScope mySet
+            let! currentCallable = (!stateRef).currentCallable
+            do! not (mySet.Contains currentCallable.FullName) |> check
+            let s1 = printNamespaceElem (QsCallable currentCallable)
+            let s2 = printScope 0 newScope
+            let newScope2 = this.ScopeTransformation newScope
```

Comment:
Can you find a better naming? If you have to start numbering your variables, the names are not descriptive enough.

Reply:
Fixed in future PR